### ISSUE TITLE
[Shared] Create shared training argument parser

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#2236]
## Summary
Create reusable training argument parsing utilities to reduce duplication across examples.

## Location
NEW: `shared/utils/training_args.mojo`

## Details
Currently each example has its own `parse_args()` function with similar patterns:
- --epochs
- --batch-size
- --lr (learning rate)
- --momentum
- --data-dir
- --weights-dir

## Proposed API

```mojo
struct TrainingConfig:
    var epochs: Int
    var batch_size: Int
    var learning_rate: Float32
    var momentum: Float32
    var data_dir: String
    var weights_dir: String

fn parse_training_args() raises -> TrainingConfig
```

## Acceptance Criteria
- [ ] `TrainingConfig` struct created
- [ ] `parse_training_args()` function implemented
- [ ] Exported from `shared/utils/__init__.mojo`
- [ ] Documentation added
- [ ] Examples can optionally adopt

## Priority
P2 - Medium

**Comment by mvillmow** [comment]
This issue has been addressed - training_args.mojo is now fully implemented with TrainingArgs struct, parse_training_args(), and parse_training_args_with_defaults() functions. All exports are in shared/utils/__init__.mojo.
